src/Pronamic.php

```diff
@@ -80,80 +80,41 @@ public static function get_submission_value( $type ) {
 				continue;
 			}
 
-			$value = array_key_exists( $tag->name, $_POST ) ? \sanitize_text_field( \wp_unslash( $_POST[ $tag->name ] ) ) : '';
+			// Submission value.
+			$value = $submission->get_posted_string( $tag->name );
 
-			$value = trim( $value );
-
-			if ( 'checkbox' === $tag->basetype ) {
-				$value = \filter_input( \INPUT_POST, $tag->name, \FILTER_DEFAULT, \FILTER_REQUIRE_ARRAY );
-			}
-
-			// Check empty value.
 			if ( empty( $value ) ) {
 				continue;
 			}
 
-			$values = (array) $value;
-
-			// Loop values.
-			foreach ( $values as $value ) {
-				/*
-				 * Try to get value from piped field values.
-				 *
-				 * @link https://contactform7.com/selectable-recipient-with-pipes/
-				 */
-				if ( $tag->pipes instanceof WPCF7_Pipes ) {
-					// Make multidimensional array with pipe options by value.
-					$options = [];
-
-					$labels = $tag->pipes->collect_befores();
-
-					foreach ( $tag->pipes->collect_afters() as $key => $after ) {
-						// Make sure array for value exists.
-						if ( ! \array_key_exists( $after, $options ) ) {
-							$options[ $after ] = [];
+			switch ( $type ) {
+				case 'amount':
+					/**
+					 * Contact Form 7 concatenates the field option value with user input for free text fields. We
+					 * are only interested in the input value as amount.
+					 *
```

Comment:
Is this complexity with 'pipes', `str_starts_with` and `substr_replace` necessary to handle the concatenate?

Reply:
I don't like it either, but without it any numbers in the option label/value will be included in the field value because of the concatenation. For example, an option `Other (minimum € 10)` and entered value of `50` will result in an amount of € 1050 instead of € 50.